### PR TITLE
Painting match

### PR DESCRIPTION
**Pull request by PokeCodec**

### CODE REVIEW

File: src/contest_painting.c [hunk 1]
```diff
@@ -384,11 +384,9 @@ static void InitContestMonPixels(u16 species, u8 whichSprite)
     }
 }
 
-#ifdef NONMATCHING
-// functionally equivalent.
 static void _InitContestMonPixels(u8 *spriteGfx, u16 *palette, u16 (*destPixels)[64][64])
 {
-    u16 tileY, tileX, pixelY, pixelX;
+    u16 tileY, tileX, pixelY, pixelX, temp;
```
Comment (GriffinRichards) [inline src/contest_painting.c:389]: Unnecessary temp

File: src/contest_painting.c [hunk 2]
```diff
@@ -399,125 +397,20 @@ static void _InitContestMonPixels(u8 *spriteGfx, u16 *palette, u16 (*destPixels)
...
                 for (pixelX = 0; pixelX < 8; pixelX++)
                 {
-                    int offset = 32 * (8 * tileY + tileX) + (pixelY * 4 + pixelX / 2);
-                    colorIndex = spriteGfx[offset];
+                    colorIndex = spriteGfx[(((tileY * 8) + tileX) * 32) + (pixelY << 2) + (pixelX >> 1)];
                     if (pixelX & 1)
                         colorIndex >>= 4;
                     else
-                        colorIndex &= 0xF;
-
```
Comment (GriffinRichards) [inline src/contest_painting.c:408]: Put this newline back, harder to read with the two conditionals squished together

**Comment by PokeCodec** [comment]
Fixed!

File: src/contest_painting.c [hunk 3]
```diff
@@ -399,125 +397,21 @@ static void _InitContestMonPixels(u8 *spriteGfx, u16 *palette, u16 (*destPixels)
...
-                        colorIndex &= 0xF;
+                        colorIndex &= 0xF; // %=16 works here too. Both match
 
-                    if (colorIndex == 0) // transparent pixel
-                        (*destPixels)[8 * tileY + pixelY][tileX * 8 + pixelX] = 0x8000;
+                    if (colorIndex == 0)   // transparent pixel
+                        (*destPixels)[((tileY * 8) + pixelY)][((tileX * 8) + pixelX)] = 0x8000;
                     else
-                        (*destPixels)[8 * tileY + pixelY][tileX * 8 + pixelX] = palette[colorIndex];
+                        (*destPixels)[((tileY * 8) + pixelY)][((tileX * 8) + pixelX)] = palette[colorIndex];
```
Comment (GriffinRichards) [inline src/contest_painting.c:409]: None of this is necessary to match either, just the changes to the `colorIndex` assignment

**Comment by PokeCodec** [comment]
All fixed!

**Pull request merged**